## This is a example of how you will be doing projects in python.
### Use markdown cells -OR- comments in cells to place AT LEAST the question number you are working on. 

3.	Import the required packages

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

4. Using the pandas `read_csv()` method, read the GDP dataset into your notebook as a DataFrame called `gdp_df`. Take a look at the first few and last few rows to familiarize yourself with what is contained in this dataset.

In [2]:
gdp_df = pd.read_csv('../data/gdp_percapita.csv')

In [3]:
gdp_df.head()

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2023  1992.424394              NaN
1     Afghanistan  2022  1981.710168              NaN
2     Afghanistan  2021  2144.166570              NaN
3     Afghanistan  2020  2769.685745              NaN
4     Afghanistan  2019  2927.245144              NaN

In [4]:
gdp_df.tail()

Country or Area  Year        Value  Value Footnotes
8214        Zimbabwe  1994  3965.730986              NaN
8215        Zimbabwe  1993  3634.750494              NaN
8216        Zimbabwe  1992  3649.891947              NaN
8217        Zimbabwe  1991  4126.405247              NaN
8218        Zimbabwe  1990  4013.299059              NaN

5. How many rows and columns are in `gdp_df`? What are the data types of each column?

In [6]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8219 entries, 0 to 8218
Data columns (total 4 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Country or Area  8219 non-null   object 
 1   Year             8219 non-null   int64  
 2   Value            8219 non-null   float64
 3   Value Footnotes  0 non-null      float64
dtypes: float64(2), int64(1), object(1)
memory usage: 257.0+ KB


8219 rows and 4 columns. Object, int, float, float

6. Drop the `Value Footnotes` column and rename the remaining three to 'Country', 'Year', and 'GDP_Per_Capita'.

In [20]:
gdp_df= gdp_df.drop(["Value Footnotes", "Country", "GDP_Per_Capita"], axis=1)

In [21]:
gdp_df=gdp_df.rename(columns={"Country or Area":"Country", "Value":"GDP_Per_Capita"})

7. How many countries have data for all years? Which countries are missing many years of data? Look at the number of observations per year. What do you notice? 

In [22]:
print(gdp_df)

          Country  Year  GDP_Per_Capita
0     Afghanistan  2023     1992.424394
1     Afghanistan  2022     1981.710168
2     Afghanistan  2021     2144.166570
3     Afghanistan  2020     2769.685745
4     Afghanistan  2019     2927.245144
...           ...   ...             ...
8214     Zimbabwe  1994     3965.730986
8215     Zimbabwe  1993     3634.750494
8216     Zimbabwe  1992     3649.891947
8217     Zimbabwe  1991     4126.405247
8218     Zimbabwe  1990     4013.299059

[8219 rows x 3 columns]


In [28]:
gdp_df['Year'].nunique()

34

In [32]:
gdp_dfYear_Count=gdp_df.groupby(['Country']).count().sort_values(by='Year')

In [34]:
gdp_dfYear_Count[gdp_dfYear_Count['Year']>33]

Year  GDP_Per_Capita
Country                                                     
Peru                                      34              34
Netherlands                               34              34
New Zealand                               34              34
Nicaragua                                 34              34
Niger                                     34              34
...                                      ...             ...
Heavily indebted poor countries (HIPC)    34              34
High income                               34              34
Honduras                                  34              34
Georgia                                   34              34
Zimbabwe                                  34              34

[228 rows x 2 columns]

In [35]:
gdp_dfYear_Count[gdp_dfYear_Count['Year']<34]

Year  GDP_Per_Capita
Country                                        
Djibouti                     11              11
Turks and Caicos Islands     13              13
Sint Maarten (Dutch part)    15              15
Faroe Islands                16              16
Kosovo                       16              16
Cayman Islands               18              18
Virgin Islands               21              21
Curaçao                     24              24
Afghanistan                  24              24
San Marino                   26              26
Montenegro                   27              27
Serbia                       29              29
West Bank and Gaza           30              30
Greenland                    32              32
Syrian Arab Republic         33              33
Somalia                      33              33
Tonga                        33              33
Bhutan                       33              33
Lebanon                      33              33

In [37]:
gdp_dfYear_Count['Year'].value_counts().sort_index()

Year
11      1
13      1
15      1
16      2
18      1
21      1
24      2
26      1
27      1
29      1
30      1
32      1
33      5
34    228
Name: count, dtype: int64

8. In this question, you're going to create some plots to show the distribution of GDP per capita for the year 2020. Go to the Python Graph Gallery (https://www.python-graph-gallery.com/) and look at the different types of plots under the Distribution section. Create a histogram, a density plot, a boxplot, and a violin plot. What do you notice when you look at these plots? How do the plots compare and what information can you get out of one type that you can't necessarily get out of the others?

In [55]:
selected_year=2020

In [56]:
gdp_df_selected_year=gdp_df[gdp_df['Year'] ==2020]

In [60]:
sns.histplot(gdp_df.Year, gdp_df.GDP_Per_Capita)

TypeError: histplot() takes from 0 to 1 positional arguments but 2 were given

9. What was the median GDP per capita value in 2020?